kubernetes/processbot/templates/processbot.yaml

```diff
@@ -54,6 +54,17 @@ spec:
         ports:
         - name: backend
           containerPort: {{ .Values.processbot.config.WEBHOOK_PORT }}
+        readinessProbe:
+          tcpSocket:
```

Comment:
By default the fallback route will return status code `404`. Would that cause this check to fail?

https://github.com/paritytech/parity-processbot/blob/bee58c73fa2a751dc8d0be3b3b1b87cf2b12d1f1/src/webhook.rs#L100

Any reason to not use the `/health` endpoint for both probes? e.g. cla-bot-2021 uses the same route for both probes https://gitlab.parity.io/parity/opstooling/cla-bot-2021/-/blob/6e1e313f9420d9805f092bff800aca6ce8ae3034/kubernetes/helm/cla-bot-2021/templates/deployment.yaml#L42

Also please ask for some review on this from DevOps

Reply:
> By default the fallback route will return status code `404`. Would that cause this check to fail?

No, the TCP probe will pass so long as _something_ is open on a given port

> Any reason to not use the /health endpoint for both probes? e.g. cla-bot-2021 uses the same route for both probes

No major reason, it just seems wrong to me to just duplicate the healthcheck here. The TCP probe will verify that the server port is open, the HTTP probe will be regularly pinged by Prom/K8s to verify the code is executing.

Reply:
Added @haikoschol and @lovelaced for a quick devops review 👍🏻

Reply:
I agree that `httpGet` should be used for HTTP services. However, in general these probes serve different purposes and can be quite tricky to get right. Otherwise they can even reduce reliability under certain conditions (some long-ish explanations [here](https://srcco.de/posts/kubernetes-liveness-probes-are-dangerous.html) and [here](https://blog.colinbreck.com/kubernetes-liveness-and-readiness-probes-revisited-how-to-avoid-shooting-yourself-in-the-other-foot/)).

The `livenessProbe` would be useful if the bot sometimes deadlocks and the pod needs to be restarted to resolve it. Has this been an issue? I can imagine it might when accessing the database. But would the `/health` endpoint actually fail in that case, given that the handler doesn't access the database?

The `readinessProbe` would be useful if processbot was running in multiple pods for scalability and there was some delay before a newly started pod should be included by the load balancer.

Having said that, in this concrete case I don't think the probes will cause any problems.

Reply:
The TCP probe is actually really useful for Polkadot as it will succeed only when the node is fully synced as such serving as a readiness check.